test/createAzureResource.test.ts

```diff
@@ -0,0 +1,103 @@
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+        ext.tree.dispose();
+    });
+
+    test('Create and Delete New Web App (Basic)', async () => {
+        const appName: string = getRandomHexString().toLowerCase();
+        await vscode.workspace.getConfiguration(constants.extensionPrefix).update('advancedCreation', false, vscode.ConfigurationTarget.Global);
+        const testInputs: string[] = [appName, 'Linux', runtimeLTS.displayName];
+        ext.ui = new TestUserInput(testInputs);
+
+        await vscode.commands.executeCommand('appService.CreateWebApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const defaultRgName: string = 'appsvc_rg_linux_centralus';
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(defaultRgName, appName);
+        assert.ok(createdApp);
+        if (deleteDefaultResourceGroup) {
+            resourceGroupsToDelete.push(defaultRgName);
+        }
+
+        ext.ui = new TestUserInput([appName, DialogResponses.deleteResponse.title, DialogResponses.yes.title]);
+        await vscode.commands.executeCommand('appService.Delete');
+        const deletedApp: WebSiteManagementModels.Site | undefined = await client.webApps.get(defaultRgName, appName);
+        assert.ifError(deletedApp); // if app was deleted, get() returns null.  assert.ifError throws if the value passed is not null/undefined
+    });
+
+    test('Create and Delete New Web App (Advanced)', async () => {
+        const resourceName: string = getRandomHexString().toLowerCase();
+        await vscode.workspace.getConfiguration(constants.extensionPrefix).update('advancedCreation', true, vscode.ConfigurationTarget.Global);
+        const testInputs: string[] = [resourceName, '$(plus) Create new resource group', resourceName, 'Linux', runtimeLTS.displayName, '$(plus) Create new App Service plan', resourceName, 'B1', 'West US'];
+        ext.ui = new TestUserInput(testInputs);
+
+        await vscode.commands.executeCommand('appService.CreateWebApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName, resourceName);
+        assert.ok(createdApp);
+        resourceGroupsToDelete.push(resourceName);
```

Comment:
You need to push the name to resourceGroupsToDelete before you try creating any resource. It could've failed to create the web app, but it might've already created the resource group or plan. These are running nightly so I want to be very careful to make sure we're cleaning up azure resources.